crates/weaver_cache/src/lib.rs

```diff
@@ -152,65 +185,438 @@ impl Cache {
+
+        create_dir_all(cache_path.as_path()).map_err(|e| Error::CacheDirNotCreated {
+            message: e.to_string(),
+        })?;
+
+        let tmp_dir = TempDir::new_in(cache_path.as_path(), "repo").map_err(|e| {
+            Error::CacheDirNotCreated {
+                message: e.to_string(),
+            }
+        })?;
+        Ok(tmp_dir)
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
-    /// Marked as ignore because we don't want to clone the repo every
-    /// time we run the tests in CI.
+    fn count_yaml_files(repo_path: &Path) -> usize {
+        let count = walkdir::WalkDir::new(repo_path)
+            .into_iter()
+            .filter_map(|e| e.ok())
+            .filter(|e| e.path().extension().map_or(false, |ext| ext == "yaml"))
+            .count();
+        count
+    }
+
     #[test]
-    #[ignore]
-    fn test_cache() {
-        let cache = Cache::try_new().unwrap();
-        let result = cache.git_repo(
-            "https://github.com/open-telemetry/semantic-conventions.git".into(),
-            Some("model".into()),
+    fn test_semconv_registry_local_repo() {
+        // A RegistryRepo created from a local folder.
+        let registry_path = RegistryPath::LocalFolder {
+            path: "../../crates/weaver_codegen_test/semconv_registry".to_owned(),
+        };
+        let repo = RegistryRepo::try_new("main", &registry_path).unwrap();
+        let repo_path = repo.path().to_path_buf();
+        assert!(repo_path.exists());
+        assert!(
+            count_yaml_files(&repo_path) > 0,
+            "There should be at least one `.yaml` file in the repo"
+        );
+        // Simulate a RegistryRepo going out of scope.
+        drop(repo);
+        // The local folder should not be deleted.
+        assert!(repo_path.exists());
+    }
+
+    fn check_archive(registry_path: RegistryPath, file_to_check: Option<&str>) {
+        let repo = RegistryRepo::try_new("main", &registry_path).unwrap();
+        let repo_path = repo.path().to_path_buf();
+        // At this point, the repo should be cloned into a temporary directory.
+        assert!(repo_path.exists());
+        assert!(
+            count_yaml_files(&repo_path) > 0,
+            "There should be at least one `.yaml` file in the repo"
         );
-        assert!(result.is_ok());
-        assert!(result.unwrap().exists());
+        if let Some(file_to_check) = file_to_check {
+            let file_path = repo_path.join(file_to_check);
+            assert!(file_path.exists());
+        }
+        // Simulate a RegistryRepo going out of scope.
+        drop(repo);
+        // The temporary directory should be deleted automatically.
+        assert!(!repo_path.exists());
+    }
+
+    #[test]
+    fn test_semconv_registry_git_repo() {
+        let registry_path = RegistryPath::GitRepo {
+            // This git repo is expected to be available.
+            url: "https://github.com/open-telemetry/semantic-conventions.git".to_owned(),
```

Comment:
One thing i'm thinking over time - we should move away from relying on remote APIs like this and instead stand up a local server that hosts the file, for stability purposes.

Not sure how hard that is now, but we can at least open a bug to track this.  

It's happened in the past that we moved/renamed repositories, don't want this test to fail then.

Reply:
Created https://github.com/open-telemetry/weaver/issues/267